examples/bond/assets/bond-dapp-stateful.py

```diff
@@ -0,0 +1,209 @@
+    epoch = Bytes("epoch")
+    total = Bytes("total")
+    current_bond = Bytes("current_bond")
+    max_amount = Bytes("max_amount")
+    creator = Bytes("creator")
+
+    # verify rekey_to and close_rem_to are set as zero_address
+    basic_checks = And(
+        # Always verify that the RekeyTo property of any transaction is set to the ZeroAddress
+        # unless the contract is specifically involved ina rekeying operation.
+        Txn.rekey_to() == Global.zero_address(),
+        Txn.close_remainder_to() == Global.zero_address(),
+        Txn.asset_close_to() == Global.zero_address()
+    )
+
+    # initialize variables
+    on_initialize = Seq([
+        App.globalPut(store_manager, Txn.application_args[0]),
+        App.globalPut(issue_price, Btoi(Txn.application_args[1])),
+        App.globalPut(nominal_price, Btoi(Txn.application_args[2])),
+        App.globalPut(maturity_date, Btoi(Txn.application_args[3])),
+        App.globalPut(coupon_value, Btoi(Txn.application_args[4])),
+        App.globalPut(epoch, Btoi(Txn.application_args[5])),
+        App.globalPut(current_bond, Btoi(Txn.application_args[6])),
+        App.globalPut(max_amount, Btoi(Txn.application_args[7])),
+        App.globalPut(creator, Txn.application_args[8]),
+        Return(Int(1))
+    ])
+
+    # only store manager can update issuer
+    update_issuer = Seq([
+        Assert(
+             And(
+                Txn.sender() == App.globalGet(store_manager),
+                basic_checks
+            )
+        ),
+        App.globalPut(issuer_address, Txn.application_args[1]),
+        Return(Int(1))
+    ])
+
+    # only bond app manager can update issue price
+    on_update_issue_price = Seq([
+        Assert(
+             And(
+                Txn.sender() == App.globalGet(store_manager),
+                basic_checks
+            )
+        ),
+        App.globalPut(issue_price, Txn.application_args[1]),
+        Return(Int(1))
+    ])
+
+    on_issue = Seq([
+        Assert(
+            And(
+                Gtxn[0].type_enum() == TxnType.AssetTransfer,
+                Gtxn[0].asset_receiver() == App.globalGet(issuer_address),
+                Gtxn[0].xfer_asset() == App.globalGet(current_bond),
+                basic_checks
+            )
+        ),
+        Return(Int(1))
+    ])
+
+    on_buy = Seq([
+        Assert(
+            And(
+                basic_checks,
+                # verify payment
+                Gtxn[0].type_enum() == TxnType.Payment,
+                # verify buying amount,
+                Gtxn[0].amount() >= Add(Mul(Gtxn[1].asset_amount(), App.globalGet(issue_price)), Gtxn[1].fee()),
+                # verify ASA transfer
+                Gtxn[1].type_enum() == TxnType.AssetTransfer,
+                Gtxn[1].xfer_asset() == App.globalGet(current_bond),
+                Gtxn[1].asset_receiver() == Gtxn[0].sender()
+            )
+        ),
+        App.globalPut(total, Add(App.globalGet(total), Gtxn[1].asset_amount())),
```

Comment:
I think we should explain the transaction composition somewhere in docs as well. i.e in this case 2nd transaction in group should be asset_transfer transaction when calling `on_buy`. What do you think?

Reply:
i have added this while we call `buy` in scripts, will copy-paste here as well. for more clarity